packages/app/src/cli/services/app-logs/helpers.ts

```diff
@@ -0,0 +1,67 @@
+import {DeveloperPlatformClient} from '../../utilities/developer-platform-client.js'
+import {AppLogsSubscribeVariables} from '../../api/graphql/subscribe_to_app_logs.js'
+import {fetch, Response} from '@shopify/cli-kit/node/http'
+import {outputDebug, outputWarn} from '@shopify/cli-kit/node/output'
+import {partnersFqdn} from '@shopify/cli-kit/node/context/fqdn'
+import {AbortError} from '@shopify/cli-kit/node/error'
+
+export const POLLING_INTERVAL_MS = 450
+export const POLLING_ERROR_RETRY_INTERVAL_MS = 5 * 1000
+export const POLLING_THROTTLE_RETRY_INTERVAL_MS = 60 * 1000
+export const ONE_MILLION = 1000000
+export const LOG_TYPE_FUNCTION_RUN = 'function_run'
+
+export const generateFetchAppLogUrl = async (
+  cursor?: string,
+  filters?: {
+    status?: string
+    source?: string
+  },
+) => {
+  const fqdn = await partnersFqdn()
+  let url = `https://${fqdn}/app_logs/poll`
+
+  if (!cursor) {
+    return url
+  }
+
+  url += `?cursor=${cursor}`
+
+  if (filters?.status) {
+    url += `&status=${filters.status}`
+  }
+  if (filters?.source) {
+    url += `&source=${filters.source}`
+  }
+
+  return url
+}
+
+export const fetchAppLogs = async (url: string, jwtToken: string): Promise<Response> => {
+  return fetch(url, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${jwtToken}`,
+    },
+  })
+}
```

Comment:
Thoughts on combining these two helpers into one? I can't think of a reason why we'd need one without the other

Reply:
Agreeded. I think we only want to separate these if we do not want dev to be able to filter the results, but partners can hack this pretty easily if they wanted.

Reply:
Sounds good. I kept them separate within the helper to keep the two things clear, but had one call the other, so the result is that anything calling this only has to make the 1 call to `fetchAppLogs` with the right params.